content/ja/docs/concepts/cluster-administration/logging.md

```diff
@@ -0,0 +1,199 @@
+---
+reviewrs:
+title: ロギングのアーキテクチャ
+content_type: concept
+weight: 60
+---
+
+<!-- overview -->
+
+アプリケーションログは、アプリケーション内で何が起こっているかを理解するのに役立ちます。ログは、問題のデバッグとクラスターアクティビティの監視に特に役立ちます。最近のほとんどのアプリケーションには、何らかののロギングメカニズムがあります。同様に、コンテナエンジンはロギングをサポートするように設計されています。コンテナ化されたアプリケーションで、最も簡単で最も採用されているロギング方法は、標準出力と標準エラーストリームへの書き込みです。
+
+ただし、コンテナエンジンまたはランタイムによって提供されるネイティブ機能は、たいていの場合、完全なロギングソリューションには十分ではありません。
+
+たとえば、コンテナがクラッシュした場合やポッドが削除された場合、またはノードが停止した場合に、アプリケーションのログにアクセスしたい場合があります。
+
+クラスターでは、ノードやポッド、またはコンテナに関係なく、ノードに個別のストレージとライフサイクルが必要です。この概念は、_クラスターレベルロギング_ と呼ばれます。
+
+<!-- body -->
+
+クラスターレベルロギングのアーキテクチャでは、ログを保存、分析、およびクエリするための個別のバックエンドが必要です。Kubernetesは、ログデータ用のネイティブストレージソリューションを提供していません。代わりに、Kubernetesに統合される多くのロギングソリューションがあります。次のセクションでは、ノードでログを処理および保存する方法について説明します。
+
+## Kubernetesでの基本的なロギング
+
+この例では、1秒に1回標準出力ストリームにテキストを書き込むコンテナを利用する、`Pod` specificationを使います。
+
+{{< codenew file="debug/counter-pod.yaml" >}}
+
+このポッドを実行するには、次のコマンドを使用します:
+
+```shell
+kubectl apply -f https://k8s.io/examples/debug/counter-pod.yaml
+```
+
+出力は次のようになります:
+
+```console
+pod/counter created
+```
+
+ログを取得するには、以下のように`kubectl logs`コマンドを使用します:
+
+```shell
+kubectl logs counter
+```
+
+出力は次のようになります:
+
+```console
+0: Mon Jan  1 00:00:00 UTC 2001
+1: Mon Jan  1 00:00:01 UTC 2001
+2: Mon Jan  1 00:00:02 UTC 2001
+...
+```
+
+コンテナの以前のインスタンスからログを取得するために、`kubectl logs --previous`を使用できます。ポッドに複数のコンテナがある場合は、コマンドにコンテナ名を追加して、アクセスするコンテナのログを指定します。詳細については、[`kubectl logs` ドキュメント](/docs/reference/generated/kubectl/kubectl-commands#logs)を参照してください。
+
+## ノードレベルでのロギング
+
+![Node level logging](/images/docs/user-guide/logging/logging-node-level.png)
+
+コンテナエンジンは、生成された出力を処理して、コンテナ化されたアプリケーションの`stdout`と`stderr`ストリームにリダイレクトします。たとえば、Dockerコンテナエンジンは、これら2つのストリームを[ロギングドライバ](https://docs.docker.com/engine/admin/logging/overvie)にリダイレクトします。ロギングドライバは、JSON形式でファイルに書き込むようにKubernetesで設定されています。
+
+{{< note >}}
+Docker JSONロギングドライバーは、各行を個別のメッセージとして扱います。Dockerロギングドライバーを使用する場合、複数行メッセージを直接サポートすることはできません。ロギングエージェントレベルあるいはそれ以上のレベルで、複数行のメッセージを処理する必要があります。
+{{< /note >}}
+
+デフォルトでは、コンテナが再起動すると、kubeletは1つの終了したコンテナをログとともに保持します。ポッドがノードから削除されると、対応する全てのコンテナが、ログとともに削除されます。
+
+ノードレベルロギングでの重要な考慮事項は、ノードで使用可能な全てのストレージをログが消費しないように、ログローテーションを実装することです。Kubernetesはログのローテーションを担当しませんが、デプロイツールでそれに対処するソリューションを構築する必要があります。たとえば、`kube-up.sh`スクリプトによってデプロイされたKubernetesクラスターには、1時間ごとに実行するように構成された[`logrotate`](https://linux.die.net/man/8/logrotate)があります。アプリケーションのログを自動的にローテーションするようにコンテナランタイムを構築することもできます。
+
+例として、[`configure-helper` script](https://github.com/kubernetes/kubernetes/blob/{{< param "githubbranch" >}}/cluster/gce/gci/configure-helper.sh)に対応するスクリプトである`kube-up.sh`が、どのようにGCPでCOSイメージのロギングを構築しているかについて、詳細な情報を見つけることができます。
```

Comment:
```suggestion
例として、[`configure-helper` script](https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/gci/configure-helper.sh)に対応するスクリプトである`kube-up.sh`が、どのようにGCPでCOSイメージのロギングを構築しているかについて、詳細な情報を見つけることができます。
```
This script is not found because  `main` branch is not used in `kubernetes/kubernetes` repository.
And `master` is used in the en page.